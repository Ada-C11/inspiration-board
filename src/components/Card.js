import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  onDeleteClick = () => {
    this.props.onDeleteClick(this.props.id)
    console.log(this.props.id)
    console.log(this.props)
    console.log(emoji.getUnicode("beer"))
  }

  render() {
    return (
      <div className="card">
        <div className="card__content">
          <h2 className="card__content-text">{this.props.text}</h2>
          <h2 className="card__content-emoji">{emoji.getUnicode(this.props.emoji)}</h2>
        </div>
        <button 
          className="card__delete" 
          onClick={this.onDeleteClick}>
          Delete
        </button>

      </div>
    )
  }
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string.isRequired,
  deleteCardCallback: PropTypes.func.isRequired,
};

export default Card;
