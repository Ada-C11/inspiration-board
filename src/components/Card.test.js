import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    const props = {
      key: 1,
      id: 1,
      text: 'You got this!',
      emoji: '',
    }

    const wrapper = shallow( <Card {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});