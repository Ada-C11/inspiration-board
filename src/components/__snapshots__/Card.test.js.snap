// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    >
       
      text
    </p>
    <p
      className="card__content-emoji"
    >
       
      üê∂
       
    </p>
    <button
      className="card__delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
